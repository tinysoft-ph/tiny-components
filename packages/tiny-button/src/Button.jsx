import React from 'react';
import PropTypes from 'prop-types';

const Button = ( {
    children,
    className,
    onClick,
} ) => {
  return (
    <button
      className={className}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.oneOf( ['default', 'primary', 'success', 'danger'] ).isRequired,
};

Button.defaultProps = {
  type: 'default',
};

export default Button;
