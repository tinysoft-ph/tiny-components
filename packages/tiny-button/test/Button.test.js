import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../src/index';

test( 'should render with type', () => {
  const component = renderer.create(
    <Button type="primary" />,
  );
  expect( component.toJSON() ).toMatchSnapshot();
} );
