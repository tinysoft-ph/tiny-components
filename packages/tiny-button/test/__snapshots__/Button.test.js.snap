// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with type 1`] = `
<button
  className={undefined}
  onClick={undefined}
/>
`;
